addons/stock_account/__init__.py

```diff
@@ -39,3 +41,30 @@ def _configure_journals(env):
             data['account.journal'] = ChartTemplate._get_stock_account_journal(template_code)
         ChartTemplate._load_data(data)
         ChartTemplate._post_load_data(template_code, company, template_data)
+
+    # if we already have quantity on hand for storable product, create an initial valuation layer
+    valued_locations = env["stock.location"].search([]).filtered(lambda loc: loc._should_be_valued())
+    qty_by_product = env["stock.quant"]._read_group(
+        [
+            ("product_id.is_storable", "=", True),
+            ("location_id", "in", valued_locations.ids),
+            ("owner_id", "=", False),
+        ],
+        groupby=["company_id", "product_id"],
+        aggregates=["quantity:sum"],
+        having=[("quantity:sum", "!=", 0)],
+    )
+    svl_vals = []
+    for company, product, quantity in qty_by_product:
+        value = quantity * product.standard_price
```

Comment:
I would skip product having `cost == 0`

Reply:
This would also result in inconsistencies, the quantity in the valuation for products with `cost == 0` will be different from the quantity on hands.
The PR is not about correcting the value, but the quantity.

However, it should be `product.with_company(company).standard_price` instead :thinking:

Reply:
@Whenrow Just tested it, and the `stock_valuation_layer` table is dropped when uninstalling the module. https://github.com/odoo/odoo/blob/8c6aefc4de5d4203d9fa20d3f7ac5860dbafbd3d/odoo/addons/base/models/ir_model.py#L361

Still, that's a good idea; I wouldn't have thought of that. :sweat_smile: 
FYI, I rebased and fixed the runbot warning about translations :)